File: LeetCode/C#/1-99/1.cs
using System;
using System.Collections.Generic;

namespace LeetCode1
{
	public class Solution {
	    public int[] TwoSum(int[] nums, int target) {
	        Dictionary<int, int> repeat = new Dictionary<int, int>();
	        int[] ans = new int[2];
	        for(int i = 0;i < nums.Length;++i){
	            if(repeat.ContainsKey(nums[i])){
	                ans[0] = repeat[nums[i]];
	                ans[1] = i;
	                break;
	            }
	            repeat[target - nums[i]] = i;
	        }
	        
	        return ans;
	    }
	}
	
	public class Program
	{
	
		public static void Main()
		{
			int[] nums = new int[]{2,7,11,15};
			Solution sol = new Solution();
			int[] ans = sol.TwoSum(nums, 9);
			foreach(int num in ans)
			{
				Console.Write(num + " ");
			}
			Console.Read();
		}
	}
}

File: LeetCode/C++/1-99/1.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> repeat;
        vector<int> ans;
        for(int i = 0;i < nums.size();++i){
            if(repeat.count(nums[i])){
                ans.push_back(repeat[nums[i]]);
                ans.push_back(i);
                break;
            }
            repeat[target - nums[i]] = i;
        }
        
        return ans;
    }
};

int main() {
	vector<int> nums{2,7,11,15};
	Solution sol;
	vector<int> ans = sol.twoSum(nums, 9);
	for(int num : ans){
		cout << num << " ";
	}
	return 0;
}
@u8989332
 